src/zenml/container_registries/aws_container_registry.py

```diff
@@ -0,0 +1,51 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+from typing import ClassVar
+
+from pydantic import validator
+
+from zenml.container_registries.base_container_registry import (
+    BaseContainerRegistry,
+)
+from zenml.enums import ContainerRegistry, StackComponentType
+
+
+class AWSContainerRegistry(BaseContainerRegistry):
+    """Class for AWS Container Registry.
+
+    Attributes:
+        uri: The URI of the container registry.
+    """
+
+    # Class Configuration
+    TYPE: ClassVar[StackComponentType] = StackComponentType.CONTAINER_REGISTRY
+    FLAVOR: ClassVar[str] = ContainerRegistry.AWS
+
+    @validator("uri")
+    def validate_aws_uri(cls, uri: str) -> None:
```

Comment:
I really like the validation with the error message giving a concrete example! One thing that might be worth considering is allowing a trailing slash and simply removing it? And probably we can even have a regex that checks if the uri ends with `amazonaws.com`, but I'm not familiar enough with the potential urls they might have